dev/tools/merge-pr.sh

```diff
@@ -49,10 +49,39 @@ ask_confirmation() {
   fi
 }
 
+curl_paginate_array() {
+  # as per https://developer.github.com/v3/guides/traversing-with-pagination/#changing-the-number-of-items-received, GitHub will never give us more than 100
+  url="$1?per_page=100"
+  # We need to process the header to get the pagination.  We have three
+  # options:
+  #
+  # 1. We can make an extra API call at the beginning to get the total
+  #    number of pages, search for a rel="last" link, and then loop
+  #    over all the pages.
+  #
+  # 2. We can ask for the header info with every single curl request,
+  #    search for a rel="next" link to follow to the next page, and
+  #    then parse out the body from the header.
+  #
+  # 3. We can keep getting pages until we get back an empty array
+  #
+  # We do (3) because it is relatively simple.
+  page=1
+  while true; do
+    response="$(curl -s "${url}&page=${page}")"
+    echo "${response}"
+    if [ "$(jq 'length' <<< "${response}")" -eq 0 ]; then # empty array, no more pages
```

Comment:
```suggestion
    if [ "$(jq 'length' <<< "$response")" -lt 100 ]; then # done
```
skips a fetch most of the time
and no need for {} here while we're at it